go/netutil/netutil.go

```diff
@@ -190,6 +191,64 @@ func ResolveIPv4Addrs(addr string) ([]string, error) {
 	return result, nil
 }
 
+func dnsLookup(host string) ([]net.IP, error) {
+	addrs, err := net.LookupHost(host)
+	if err != nil {
+		return nil, fmt.Errorf("Error looking up dns name [%v]: (%v)", host, err)
+	}
+	naddr := make([]net.IP, len(addrs))
+	for i, a := range addrs {
+		naddr[i] = net.ParseIP(a)
+	}
+	sort.Slice(naddr, func(i, j int) bool {
+		return bytes.Compare(naddr[i], naddr[j]) < 0
+	})
+	return naddr, nil
+}
+
+// DNSTracker is a closure that persists state for
+//  tracking changes in the DNS resolution of a target dns name
+//  returns true if the DNS name resolution has changed
+//  If there is a lookup problem, we pretend nothing has changed
+func DNSTracker(host string) func() (bool, error) {
```

Comment:
shouldn't we fail if the host we get is empty? looks to me like we would just always return false from the returned function

Reply:
That is by design,  we would not refresh the DNS in that case;  and accordingly some type of DNS issue that results in an empty record would not result in us reopening the pool, possibly ending up with a non-reachable backend DB.